tests/test_filtsmooth/test_gaussfiltsmooth/test_kalman.py

```diff
@@ -19,30 +21,89 @@ def problem():
     return ornstein_uhlenbeck()
 
 
+# Predictions
+
+
 @pytest.fixture
-def update():
-    """The usual Kalman update.
+def predict():
+    """Classical prediction."""
+    return pnfs.predict_via_transition
+
+
+@pytest.fixture
+def predict():
+    """Square-root prediction."""
+    return pnfs.predict_sqrt
+
+
+# Measures
+
+
+@pytest.fixture
+def measure():
+    """Classical measure."""
+    return pnfs.measure_via_transition
+
+
+@pytest.fixture
+def measure():
+    """Square-root measure."""
+    return pnfs.measure_sqrt
+
+
+# Updates
+
 
-    Yields Kalman filter.
-    """
+@pytest.fixture
+def update():
+    """Classical update."""
     return pnfs.update_classic
 
 
 @pytest.fixture
 def update():
-    """Iterated classical update.
+    """Square-root update."""
+    return pnfs.update_sqrt
+
 
-    Yields I(E/U)KF depending on the approximate measurement model.
-    """
+@pytest.fixture
+def update():
+    """Iterated classical update."""
     stopcrit = pnfs.StoppingCriterion()
     return pnfs.iterate_update(pnfs.update_classic, stopcrit=stopcrit)
 
 
+# Smoothing steps
+
+
+@pytest.fixture
+def smooth_step():
+    """Classical smoothing step."""
+    return pnfs.rts_smooth_step_classic
+
+
 @pytest.fixture
-def kalman(problem, update):
+def smooth_step():
+    """Square-root smoothing step."""
+    return pnfs.rts_smooth_step_sqrt
+
+
+# Test setup
+
+
+@pytest.fixture
+def kalman(problem, predict, measure, update, smooth_step):
```

Comment:
Are all 2^4 combinations of predict, measure, update, and smoothstep are tested here, for each problem?

Reply:
(just wondering about how things work, no need to change anything in any case^^)